### PR TITLE
Ensure errors are emitted when ts:precompile fails

### PR DESCRIPTION
**Pull request by dfreeman**
As noted both in Discord and in #973, we currently don't emit any of `tsc`'s output when it fails during `ts:precompile`. This ensures we capture and re-emit any output when the command exits with a nonzero code, and adds a test to make sure we don't regress in the future.

### CODE REVIEW

**Comment by dfreeman** [comment]
Floating-deps tests should start passing again once https://github.com/zloirock/core-js/issues/710 is resolved

**Comment by chriskrycho** [comment]
Updated from master, rerunning CI now. Also reviewing!

File: ts/lib/commands/precompile.ts [hunk 1]
```diff
@@ -25,18 +25,25 @@ export default command({
...
+        '--noEmit', 'false',
+        '--rootDir', rootDir || this.project.root,
+        '--isolatedModules', 'false',
+        '--declaration',
+        '--declarationDir', outDir,
+        '--emitDeclarationOnly',
+        '--pretty', 'true',
+      ], {
+        preferLocal: true,
+        all: true,
```
Comment (chriskrycho) [inline ts/lib/commands/precompile.ts]: Would you mind adding a comment here explaining why we want this? Will make maintenance easier. (I just went and read the `execa` docs to figure it out, figure it'll save us time in the future from doing the same.)

Comment (dfreeman) [reply ts/lib/commands/precompile.ts]: Done! `execa` is one of those that I've used enough places that I've lost any good sense of which options are self-describing and which aren't 🙃

Comment (chriskrycho) [reply ts/lib/commands/precompile.ts]: I sympathize!

**Comment by ef4** [comment]
Thanks, I just hit this bug and was about to make the same PR when I saw it's already fixed on master.

**Comment by jamescdavis** [comment]
Released in [3.1.2](https://github.com/typed-ember/ember-cli-typescript/releases/tag/v3.1.2)

**Pull request merged**